kotlin/src/main/kotlin/com/iamdorra/gameoflife/GridManager.kt

```diff
@@ -0,0 +1,74 @@
+package com.iamdorra.gameoflife
+
+internal class GridManager
+{
+    fun getNewState(grid: List<List<Cell>>): List<List<Cell>>
+    {
+        var newGrid = (0..grid.size).map { (0..grid[0].size).map { Cell(false) } }
```

Comment:
Ici je suis partagé. En C#, justement, on construit newGrid avec 2 boucles. A priori, refactorer les 2 boucles en map n'est pas le but du kata, alors ça me semble OK de laisser les 2 map. Je me trompe peut-être.

Reply:
L'idée est d'avoir un tableau à deux dimensions. si on doit avoir deux maps pour le faire why not.
Ils factoriseront comme ils voudront après ;)

Reply:
Comment tu ferais en Kotlin du coup le double boucle ? Avec deux ranges et deux foreachs ?

Reply:
Du coup Array = List ? Peu d'impact pour le kata ?
Et pareil que @HadrienMP j'ai bloqué à l'initialisation de l'array 2D sans les map

Reply:
Oui c'est très moche, c'est parfait. C'est d'autant plus moche que ça justifie maintenant l'utilisation de var plutôt que val pour newGrid 🤣